From b5cb8ba0a238774cba5c653a79a0471bbe101368 Mon Sep 17 00:00:00 2001
From: Masato TOYOSHIMA <phoepsilonix@phoepsilonix.love>
Date: Sat, 22 Jul 2023 20:12:28 +0900
Subject: [PATCH] In the 'pacman-key --populate' command, if no explicit
 specification is made, meaning no keyring name is provided, all keys present
 in the 'keyrings' folder will be loaded.

---
 pacman-init.service | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/sd/pacman-init.service b/data/sd/pacman-init.service
new file mode 100644
index 0000000..f9fb4a1
--- /dev/null
+++ b/data/sd/pacman-init.service
@@ -0,0 +1,16 @@
+[Unit]
+Description=Initialize Pacman keyring
+Wants=haveged.service
+After=haveged.service
+Requires=etc-pacman.d-gnupg.mount
+After=etc-pacman.d-gnupg.mount
+
+[Service]
+Type=oneshot
+RemainAfterExit=yes
+ExecStart=/usr/bin/pacman-key --init
+ExecStart=/usr/bin/pacman-key --populate
+
+[Install]
+WantedBy=multi-user.target
+
diff --git a/data/rc/pacman-init.in b/data/rc/pacman-init.in
index 3459cf6..cef6521 100755
--- a/data/rc/pacman-init.in
+++ b/data/rc/pacman-init.in
@@ -12,6 +12,6 @@ depend() {
 start() {
 	ebegin "Running pacman initialization"
 	/usr/bin/pacman-key --init &> /dev/null
-	/usr/bin/pacman-key --populate archlinux manjaro &> /dev/null
+	/usr/bin/pacman-key --populate &> /dev/null
 	eend $?
 }
